Grades.java

```diff
@@ -10,9 +10,54 @@
  * 2. The lowest grade and its index (something like: the student #3 got the lowest grade and it's 13).
  * 3. The average of the grades
  */
+
+package hw2arrays;
+
+import java.util.Scanner;
+
 public class Grades {
 
-    public static void main(String[] args) {
-        // enter your code here :)
-    }
+	public static void main(String[] args) {
+		Scanner in = new Scanner(System.in);
+		
+		System.out.print("Enter number of students:");
+		String outMessage;
+		byte numOfStud = in.nextByte();
+		byte[] grades = new byte[numOfStud]; 
+		
+		for (byte i = 0; i < grades.length; i++) {
+			System.out.print("Enter grade for student #" + i + ":");
+			byte tmpGrade = in.nextByte();
+			if(tmpGrade >= 0 && tmpGrade <= 100) {
+				grades[i] = tmpGrade;
+			} else {
+				System.out.println("Error, please enter number between 0 and 100!");
+				i--;
+			}
+		}
+	
+		byte maxGrade = grades[0];
+		byte maxGradeIndex = 0;
+		
+		byte minGrade = grades[0];
+		byte minGradeIndex = 0;
+		
+		short sum = grades[0];
+		
+		for (byte i = 1; i < grades.length; i++) {
+			if(grades[i] > maxGrade) {
+				maxGrade = grades[i];
+				maxGradeIndex = i;
+			}
+			if(grades[i] < minGrade) {
+				minGrade = grades[i];
+				minGradeIndex = i;
+			}
+			sum += grades[i];
+		}
+		outMessage = "The student #" + maxGradeIndex + " got the highest grade and it's " + maxGrade + ".\n";
+		outMessage += "The student #" + minGradeIndex + " got the lowest grade and it's " + minGrade + ".\n";
+		outMessage += "The average of the grades it's " + (float)(sum/grades.length) + ".\n";
```

Comment:
The casting operates on the whole expression "sum/grades.length" because it's wrapped with "( )", which means the casting will happen AFTER the division. That's why you need to do the casting before the division: ((float)sum) / ((float)grades.length). Actually, it's enough to cast only one of these expressions.. we'll talk about this next time I hope

Reply:
Ok, thank's, yes, it's work correctly also in this way, casting for division result: ((float)sum/grades.length)